hw_diag/diagnostics/gateway_diagnostics.py

```diff
@@ -0,0 +1,26 @@
+from hm_pyhelper.diagnostics import DiagnosticsReport
+from hm_pyhelper.diagnostics.diagnostic import Diagnostic
+from hw_diag.utilities.miner import fetch_miner_data
+from hm_pyhelper.miner_param import LOGGER
+import grpc
+
+
+class GatewayDiagnostic(Diagnostic):
+    KEY = 'GATEWAY'
+    FRIENDLY_NAME = "gatewayrs"
+
+    def __init__(self):
+        super(GatewayDiagnostic, self).__init__(self.KEY, self.FRIENDLY_NAME)
+
+    def perform_test(self, diagnostics_report: DiagnosticsReport) -> None:
+        diagnostics = {}
+        try:
+            fetch_miner_data(diagnostics)
+        except grpc.RpcError as err:
+            LOGGER.error(f"rpc error: {err}")
+            LOGGER.exception(err)
+            diagnostics_report.record_failure(err, self)
+        except Exception as err:
+            LOGGER.exception(err)
+            diagnostics_report.record_failure(err, self)
+        diagnostics_report.record_result(diagnostics, self)
```

Comment:
This pattern of scoping all of this data under the `gatewayrs` key makes perfect sense, but wasn't what I was expecting. I was expecting this [wrapper pattern](https://github.com/NebraLtd/hm-diag/blob/24240cc4088732536dacd129cdd99862f57158f3/hw_diag/diagnostics/key_diagnostics.py#L41) where several keys are added to the top level of the DiagnosticsReport. @kashifpk @NebraLtd/device-team what do you think is better?

Reply:
I'm ok with both approaches. Just a note that whichever we select should be uniform across all diagnostic reports so the code is easier to follow.

Reply:
I find grouping simpler to read but it has one disadvantage. The group and child keys have to be checked separately for validity.

Reply:
flattened it.